contracts/VioletID.sol

```diff
@@ -142,10 +142,39 @@ contract VioletID is
     function grantStatuses(
         address account,
         uint256 statusCombinationId
+    ) public override onlyRole(ADMIN_ROLE) whenNotPaused {
+        _updateMultipleStatuses(account, statusCombinationId);
+    }
+
+    /**
+     * @dev See {IVioletID-setStatuses}
+     *
+     * Only callable if contract is not paused
+     * Only callable by ADMIN_ROLE
+     */
+    function setStatuses(
+        address account,
+        uint256 statusCombinationId
     ) public override onlyRole(ADMIN_ROLE) whenNotPaused {
         _setMultipleStatuses(account, statusCombinationId);
     }
 
+    /**
+     * @dev See {IVioletID-batchSetStatuses}
+     *
+     * Only callable if contract is not paused
+     * Only callable by ADMIN_ROLE
+     */
+    function batchSetStatuses(
+        address[] calldata accountArray,
+        uint256[] calldata statusCombinationIdArray
+    ) public override onlyRole(ADMIN_ROLE) whenNotPaused {
+        require(accountArray.length == statusCombinationIdArray.length, "AccountArray length mismatch");
+        for (uint256 i = 0; i < accountArray.length; i++) {
```

Comment:
We should cache the array length and use unchecked for gas savings https://hackmd.io/@totomanov/gas-optimization-loops

Reply:
I added only the i++ operation as unchecked as we cannot be sure that the array passed by the user will not result in an overflow, therefore the i < length condition should still be checked